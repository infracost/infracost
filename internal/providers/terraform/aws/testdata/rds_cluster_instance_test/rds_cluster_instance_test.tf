provider "aws" {
  region                      = "us-east-1"
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true
  skip_get_ec2_platforms      = true
  skip_region_validation      = true
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
}

resource "aws_rds_cluster" "default" {
  cluster_identifier = "aurora-cluster-demo"
  availability_zones = ["us-east-1a", "us-east-1b", "us-east-1c"]
  database_name      = "mydb"
  master_username    = "foo"
  master_password    = "barbut8chars"
}

resource "aws_rds_cluster_instance" "cluster_instance" {
  identifier         = "aurora-cluster-demo"
  cluster_identifier = aws_rds_cluster.default.id
  instance_class     = "db.r4.large"
  engine             = aws_rds_cluster.default.engine
  engine_version     = aws_rds_cluster.default.engine_version
}

resource "aws_rds_cluster" "default_t3" {
  cluster_identifier = "aurora-cluster-demo"
  availability_zones = ["us-east-1a", "us-east-1b", "us-east-1c"]
  database_name      = "mydb"
  master_username    = "foo"
  master_password    = "barbut8chars"
}

resource "aws_rds_cluster_instance" "cluster_instance_t3" {
  identifier         = "aurora-cluster-demo"
  cluster_identifier = aws_rds_cluster.default.id
  instance_class     = "db.t3.medium"
  engine             = aws_rds_cluster.default.engine
  engine_version     = aws_rds_cluster.default.engine_version
}

resource "aws_rds_cluster_instance" "cluster_instance_performance_insights" {
  identifier                            = "aurora-cluster-demo"
  cluster_identifier                    = aws_rds_cluster.default.id
  instance_class                        = "db.r4.8xlarge"
  engine                                = aws_rds_cluster.default.engine
  engine_version                        = aws_rds_cluster.default.engine_version
  performance_insights_enabled          = true
  performance_insights_retention_period = 731
}

resource "aws_rds_cluster_instance" "cluster_instance_performance_insights_with_usage" {
  identifier                            = "aurora-cluster-demo"
  cluster_identifier                    = aws_rds_cluster.default.id
  instance_class                        = "db.t3.large"
  engine                                = aws_rds_cluster.default.engine
  engine_version                        = aws_rds_cluster.default.engine_version
  performance_insights_enabled          = true
  performance_insights_retention_period = 731
}

resource "aws_rds_cluster_instance" "cluster_instance_1yr_no_upfront" {
  identifier         = "aurora-cluster-demo"
  cluster_identifier = aws_rds_cluster.default.id
  instance_class     = "db.r4.large"
  engine             = aws_rds_cluster.default.engine
  engine_version     = aws_rds_cluster.default.engine_version
}

resource "aws_rds_cluster_instance" "cluster_instance_1yr_partial_upfront" {
  identifier         = "aurora-cluster-demo"
  cluster_identifier = aws_rds_cluster.default.id
  instance_class     = "db.r4.large"
  engine             = aws_rds_cluster.default.engine
  engine_version     = aws_rds_cluster.default.engine_version
}

resource "aws_rds_cluster_instance" "cluster_instance_1yr_all_upfront" {
  identifier         = "aurora-cluster-demo"
  cluster_identifier = aws_rds_cluster.default.id
  instance_class     = "db.r4.large"
  engine             = aws_rds_cluster.default.engine
  engine_version     = aws_rds_cluster.default.engine_version
}

resource "aws_rds_cluster_instance" "cluster_instance_3yr_partial_upfront" {
  identifier         = "aurora-cluster-demo"
  cluster_identifier = aws_rds_cluster.default.id
  instance_class     = "db.r4.large"
  engine             = aws_rds_cluster.default.engine
  engine_version     = aws_rds_cluster.default.engine_version
}

resource "aws_rds_cluster_instance" "cluster_instance_3yr_all_upfront" {
  identifier         = "aurora-cluster-demo"
  cluster_identifier = aws_rds_cluster.default.id
  instance_class     = "db.t3.medium"
  engine             = aws_rds_cluster.default.engine
  engine_version     = aws_rds_cluster.default.engine_version
}
