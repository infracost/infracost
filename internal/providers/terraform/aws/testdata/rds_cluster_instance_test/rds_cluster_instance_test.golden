
 Name                                                                               Monthly Qty  Unit                        Monthly Cost 
                                                                                                                                          
 aws_rds_cluster.default                                                                                                                  
 ├─ Storage                                                                 Monthly cost depends on usage: $0.10 per GB                   
 ├─ I/O requests                                                            Monthly cost depends on usage: $0.20 per 1M requests          
 ├─ Backtrack                                                               Monthly cost depends on usage: $0.012 per 1M change-records   
 └─ Snapshot export                                                         Monthly cost depends on usage: $0.01 per GB                   
                                                                                                                                          
 aws_rds_cluster.default_t3                                                                                                               
 ├─ Storage                                                                 Monthly cost depends on usage: $0.10 per GB                   
 ├─ I/O requests                                                            Monthly cost depends on usage: $0.20 per 1M requests          
 ├─ Backtrack                                                               Monthly cost depends on usage: $0.012 per 1M change-records   
 └─ Snapshot export                                                         Monthly cost depends on usage: $0.01 per GB                   
                                                                                                                                          
 aws_rds_cluster_instance.cluster_instance                                                                                                
 └─ Database instance (on-demand, db.r4.large)                                              730  hours                            $211.70 
                                                                                                                                          
 aws_rds_cluster_instance.cluster_instance_1yr_all_upfront                                                                                
 └─ Database instance (reserved, db.r4.large)                                               730  hours                              $0.00 
                                                                                                                                          
 aws_rds_cluster_instance.cluster_instance_1yr_no_upfront                                                                                 
 └─ Database instance (reserved, db.r4.large)                                               730  hours                            $138.70 
                                                                                                                                          
 aws_rds_cluster_instance.cluster_instance_1yr_partial_upfront                                                                            
 └─ Database instance (reserved, db.r4.large)                                               730  hours                             $65.70 
                                                                                                                                          
 aws_rds_cluster_instance.cluster_instance_3yr_all_upfront                                                                                
 └─ Database instance (reserved, db.t3.medium)                                              730  hours                              $0.00 
                                                                                                                                          
 aws_rds_cluster_instance.cluster_instance_3yr_partial_upfront                                                                            
 └─ Database instance (reserved, db.r4.large)                                               730  hours                             $43.80 
                                                                                                                                          
 aws_rds_cluster_instance.cluster_instance_performance_insights                                                                           
 ├─ Database instance (on-demand, db.r4.8xlarge)                                            730  hours                          $3,387.20 
 ├─ Performance Insights Long Term Retention (db.r4.8xlarge)                                 32  vCPU-month                       $165.89 
 └─ Performance Insights API                                                Monthly cost depends on usage: $0.01 per 1000 requests        
                                                                                                                                          
 aws_rds_cluster_instance.cluster_instance_performance_insights_with_usage                                                                
 ├─ Database instance (on-demand, db.t3.large)                                              730  hours                            $119.72 
 ├─ CPU credits                                                                              60  vCPU-hours                         $5.40 
 ├─ Performance Insights Long Term Retention (db.t3.large)                                    2  vCPU-month                         $5.90 
 └─ Performance Insights API                                                             12.345  1000 requests                      $0.12 
                                                                                                                                          
 aws_rds_cluster_instance.cluster_instance_t3                                                                                             
 ├─ Database instance (on-demand, db.t3.medium)                                             730  hours                             $59.86 
 └─ CPU credits                                                                              48  vCPU-hours                         $4.32 
                                                                                                                                          
 OVERALL TOTAL                                                                                                                  $4,208.32 
──────────────────────────────────
11 cloud resources were detected:
∙ 11 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file